### PR TITLE
feat(genai): add GenAI tab shell to span detail panel

### PR DESCRIPTION
**Pull request by swetalin-10**
## Which problem is this PR solving?

Part of #3820.

The span detail panel shows a flat accordion of raw attributes and events. For GenAI spans that carry dozens of `gen_ai.*` keys, token counts and prompt content, the raw list makes it hard to find what matters. This PR adds a tab layout so GenAI-specific information gets its own dedicated view instead of being buried in the generic attribute table.

## Description of the changes

Adds a two-tab layout to the span detail panel:

- **Details tab** — existing accordion content, unchanged for all spans; this is the default active tab
- **GenAI tab** — appears only when the span has at least one `gen_ai.*` attribute (checked via `isGenAISpan`)

For all other spans the layout is exactly as before — no tab bar, content rendered directly.

The GenAI tab body is a placeholder stub in this PR. Token counts, model info and prompt/tool-I/O rendering follow in the next PRs.

**New files:**
- `src/utils/genai/index.ts` — `isGenAISpan(span)` helper; checks `span.attributes` for any `gen_ai.*` key prefix. Note: #3856 adds `utils/genai/detect.ts` with a richer `classifySpan`; once that lands, `isGenAISpan` here will be re-expressed as `classifySpan(span) !== 'STANDARD'` and `index.ts` will become a thin re-export.
- `src/components/TracePage/TraceTimelineViewer/SpanDetail/GenAITab/index.tsx` — placeholder stub for the GenAI tab content

**Modified:**
- `SpanDetail/index.tsx` — wraps existing detail content in antd `<Tabs>`, conditionally adds GenAI tab when `showGenAITab` is true

## How was this change tested?

- `npm test -w packages/jaeger-ui -- SpanDetail/index` — 13 tests pass (9 existing + 4 new for tab show/hide/default-key behavior)
- `npm test -w packages/jaeger-ui -- utils/genai` — 5 tests pass covering `isGenAISpan` edge cases
- CI lint-build and unit-tests checks pass

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/main/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [x] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully

## AI Usage in this PR (choose one)
See [AI Usage Policy](https://github.com/jaegertracing/jaeger/blob/main/CONTRIBUTING_GUIDELINES.md#ai-usage-policy).
- [ ] **None**: No AI tools were used in creating this PR
- [ ] **Light**: AI provided minor assistance (formatting, simple suggestions)
- [x] **Moderate**: AI helped with code generation or debugging specific parts
- [ ] **Heavy**: AI generated most or all of the code changes

### CODE REVIEW

**Comment by swetalin-10** [comment]
The `check-label` check is failing because no `changelog:*` label has been applied. This is a feature addition so `changelog:feature` fits. Could a maintainer add it?

Also flagging for reviewers: `utils/genai/index.ts` introduced here will overlap with the `utils/genai/detect.ts` from #3856. The plan once #3856 merges is to remove `index.ts` and rewrite the import as `classifySpan(span) !== 'STANDARD'` from `detect.ts`.

File: packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/index.tsx [hunk 1]
```diff
@@ -97,72 +177,7 @@ export default function SpanDetail(props: SpanDetailProps) {
...
-          <CopyIcon
-            copyText={deepLinkCopyText}
-            icon={<IoLinkOutline />}
-            placement="topRight"
-            tooltipTitle="Copy deep link to this span"
-            buttonText="Copy"
-          />
-        </small>
-      </div>
+      <Tabs defaultActiveKey={showGenAITab ? 'genai' : 'details'} items={tabItems} />
```
[the remark this answers is not in the document]
Comment (swetalin-10) [reply packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/index.tsx]: Fixed — non-GenAI spans now render `detailsContent` directly with no `<Tabs>` wrapper at all, so there is no tab bar and no visual change for standard spans. `<Tabs>` is only rendered when `showGenAITab` is true.

[the remark this answers is not in the document]
Comment (swetalin-10) [reply packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/index.tsx]: Fixed — added `key={span.spanID}` on the `<Tabs>` component. When the active span changes and `SpanDetail` stays mounted, React remounts `Tabs` because its `key` changes, which resets the uncontrolled state and applies `defaultActiveKey="genai"` correctly for each new GenAI span.

[the remark this answers is not in the document]
Comment (swetalin-10) [reply packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/index.tsx]: Fixed — non-GenAI spans now render detailsContent directly with no <Tabs> wrapper at all, so there is no tab bar and no visual change for standard spans. <Tabs> is only rendered when showGenAITab is true.

[the remark this answers is not in the document]
Comment (swetalin-10) [reply packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/index.tsx]: Fixed — added key={span.spanID} on the <Tabs> component. When the active span changes and SpanDetail stays mounted, React remounts Tabs because its key changes, which resets the uncontrolled state and applies defaultActiveKey="genai" correctly for each new GenAI span.

File: packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/index.tsx [hunk 2]
```diff
@@ -97,72 +172,18 @@ export default function SpanDetail(props: SpanDetailProps) {
...
-            placement="topRight"
-            tooltipTitle="Copy deep link to this span"
-            buttonText="Copy"
-          />
-        </small>
-      </div>
+      {showGenAITab ? (
+        <Tabs
+          key={span.spanID}
+          defaultActiveKey="genai"
```
Comment (anshgoyalevil) [inline packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/index.tsx]: I think the default tab should not be the `genai` until we have concrete UI in this panel.

Comment (swetalin-10) [reply packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/index.tsx]: Good point. Changed defaultActiveKey to 'details' so the Details tab opens by default. The GenAI tab is still visible and accessible but no longer pre-selected until the panel has concrete UI. Updated the tests accordingly.

**Review by yurishkuro — COMMENTED** [review]
The `showGenAITab` condition currently gates on `useJaegerAssistantEnabled()`, which checks `backendCapabilities.aiAssistant`. That flag controls whether the AI chatbot/sidecar is reachable — it has nothing to do with displaying structured GenAI span attributes (model, tokens, prompts).

The condition should simply be:

```ts
const showGenAITab = isGenAISpan(span);
```

The `aiAssistant` capability is for the conversational assistant feature, not for richer span detail rendering. A deployment without the AI sidecar should still show the GenAI tab for spans that carry `gen_ai.*` attributes.

**Review by yurishkuro — COMMENTED** [review]
Minor: the PR description says "When both conditions are met the GenAI tab is the default active tab" but the code uses `defaultActiveKey="details"` and the test verifies Details is the default. The description should be updated to match.

**Comment by swetalin-10** [comment]
Fixed both points: Removed the `useJaegerAssistantEnabled()` gate - `showGenAITab` is now simply `isGenAISpan(span)`. The `aiAssistant` capability is for the conversational sidecar, not for rendering structured `gen_ai.*` attributes. Updated the PR description - removed the reference to `backendCapabilities.aiAssistant` as a condition, and corrected the wording about the default tab (Details is the default, not GenAI).

File: packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/index.tsx [hunk 3]
```diff
@@ -2,17 +2,19 @@
...
 import AccordionAttributes from './AccordionAttributes';
 import AccordionEvents from './AccordionEvents';
 import AccordionLinks from './AccordionLinks';
 import AccordionText from './AccordionText';
 import DetailState from './DetailState';
+import GenAITab from './GenAITab';
 import { formatDuration, formatDurationCompact } from '../utils';
 import CopyIcon from '../../../common/CopyIcon';
 import LabeledList from '../../../common/LabeledList';
+import { isGenAISpan } from '../../../../utils/genai';
```
[the remark this answers is not in the document]
Comment (swetalin-10) [reply packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/index.tsx:17]: Thanks for the review. This comment is based on the old PR description, which has since been updated to match the code. The maintainer @yurishkuro explicitly asked to remove the backendCapabilities.aiAssistant gate — that flag controls the conversational AI sidecar, not structured gen_ai.* span attribute rendering. The condition is intentionally just isGenAISpan(span).

File: packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/index.tsx [hunk 4]
```diff
@@ -86,6 +88,77 @@ export default function SpanDetail(props: SpanDetailProps) {
   ];
   const deepLinkCopyText = `${window.location.origin}${window.location.pathname}?uiFind=${span.spanID}`;
 
+  const showGenAITab = isGenAISpan(span);
+
```
[the remark this answers is not in the document]
Comment (swetalin-10) [reply packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/index.tsx:92]: Same as above — the aiAssistant gate was intentionally removed per @yurishkuro's review. showGenAITab = isGenAISpan(span) is the correct condition.

File: packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/index.tsx [hunk 5]
```diff
@@ -97,72 +170,18 @@ export default function SpanDetail(props: SpanDetailProps) {
...
-            placement="topRight"
-            tooltipTitle="Copy deep link to this span"
-            buttonText="Copy"
-          />
-        </small>
-      </div>
+      {showGenAITab ? (
+        <Tabs
+          key={span.spanID}
+          defaultActiveKey="details"
```
[the remark this answers is not in the document]
Comment (swetalin-10) [reply packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/index.tsx:176]: defaultActiveKey="details" is intentional — @anshgoyalevil asked not to default to the GenAI tab until the panel has concrete UI. The PR description has been updated to reflect this.

File: packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/index.test.jsx [hunk 6]
```diff
@@ -285,4 +298,36 @@ describe('<SpanDetail>', () => {
...
+      expect(screen.queryByRole('tab')).not.toBeInTheDocument();
+      expect(screen.getByTestId('accordian-keyvalues-tags')).toBeInTheDocument();
+    });
+
+    it('defaults to the Details tab for a GenAI span', () => {
+      isGenAISpanMock.mockReturnValue(true);
+      render(<SpanDetail {...props} />);
+      expect(screen.getByRole('tab', { name: 'Details' })).toHaveAttribute('aria-selected', 'true');
+      expect(screen.getByRole('tab', { name: 'GenAI' })).toHaveAttribute('aria-selected', 'false');
+    });
```
[the remark this answers is not in the document]
Comment (swetalin-10) [reply packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/index.test.jsx:331]: The test correctly reflects the intended behavior. defaultActiveKey="details" is by design per @anshgoyalevil's review — the Details tab should be the default until the GenAI panel has real content. The PR description has been updated to match.

**Review by yurishkuro — APPROVED** [review]
